L1Trigger/L1THGCal/plugins/be_algorithms/HGCClusterAlgo.cc

```diff
@@ -41,9 +43,13 @@ class HGCClusterAlgo : public Algorithm<FECODEC>
         trgcell_product_( new l1t::HGCalTriggerCellBxCollection ),
         cluster_product_( new l1t::HGCalClusterBxCollection ),
         multicluster_product_( new l1t::HGCalMulticlusterBxCollection ),
+        towerMap2Dcluster_product_( new l1t::HGCalClusterBxCollection ),
```

Comment:
I guess the TC calibration should be put in a separate BE algo as well, since the outputs would be used by both HGCClusterAlgo and HGCTowerAlgo, right?

Reply:
It cannot be done currently because BE algos are independent from each other. The output of one BE algo cannot be used as input of other algos. So the TC calibration would need to be duplicated (actually, the call to the calibration is duplicated but the calibration code is not).
But a restructuration of the code is currently in progress (see there slides [here](https://indico.cern.ch/event/624663/contributions/2525557/attachments/1431967/2200062/17.03.21_emulator.postTDR.pdf)), which will (among other things) allow to apply the TC calibration only once in the front-end classes.